src/content/blog/astro-530.mdx

```diff
@@ -0,0 +1,130 @@
+
+## Automatic session storage setup
+
+Astro 5.1 introduced the experimental [session storage API](https://docs.astro.build/en/reference/experimental-flags/sessions/), which allows developers to easily manage user sessions on the server. The feature supports dozens of different backend storage drivers, powered by [unstorage](https://unstorage.unjs.io).
+
+Previously you needed to configure a session storage driver in your `astro.config.mjs`, but now Astro will automatically set up the session storage for you when using a supported adapter.
+
+We're starting with support for the `@astrojs/node` adapter, which uses the filesystem for storage, and `@astrojs/netlify`, which uses Netlify Blobs. We hope to add default driver support for more adapters in the near future, where the hosting platforms include a storage API. In all cases, you can still manually configure your own session storage driver if a default one does not exist or you prefer to use a different one.
+
+If you haven't experimented with the session storage API yet, you can enable this feature by setting the `experimental.session` flag  to `true` in your `astro.config.mjs`:
+
+```js title="astro.config.mjs" ins={6}
+  {
+    adapter: node({
+      mode: "standalone",
+    }),
+    experimental: {
+      session: true,
+    },
+  }
+```
+
+If you are already using the session feature, you will need to update your configuration for the new syntax with this breaking change.
+
+The `experimental.session` option is now a boolean flag that you can set `true` and enjoy a default session storage driver with a supported adapter, no further configuration needed!
+
+Session configuration itself has moved to a top-level `session` key. This allows you to configure a non-default driver, as well as other options such as a `base` prefix or expiration `ttl`.
+
+```js title="astro.config.mjs" ins={9,11-13} del={6-8}
+  {
+    adapter: node({
+      mode: "standalone",
+    }),
+    experimental: {
+        session: {
+            driver: "upstash",
+        },
+        session: true,
+    },
+    session: {
+        driver: "upstash",
+    },
+  }
+```
+
+For more details and examples, see the [experimental session docs](https://docs.astro.build/en/reference/experimental-flags/sessions/).
+
+## More control over Netlify bundling
+
+When Astro builds your site to deploy to a serverless platform like Netlify or Vercel, it needs to bundle all the files and dependencies needed to run your site. The Netlify and Vercel adapters use [the same library for this](https://github.com/vercel/nft), which is usually good at working out what's needed. It doesn't always get this right though: sometimes it misses a file that you need, particularly if it's a static file that you are loading with filesystem functions. Other times it includes files that you don't need, which can slow down your build and increase your bundle size, particularly if it's something like a large binary.
```

Comment:
```suggestion
When Astro builds your site to deploy to a serverless platform like Netlify or Vercel, it needs to bundle all the files and dependencies needed to run your site. The Netlify and Vercel adapters use [the same library for this](https://github.com/vercel/nft), which is usually good at working out what's needed.

It doesn't always get this right though: sometimes it misses a file that you need, particularly if it's a static file that you are loading with filesystem functions. Other times it includes files that you don't need, which can slow down your build and increase your bundle size, particularly if it's something like a large binary.
```
Just splitting up into two paragraphs, but otherwise great!